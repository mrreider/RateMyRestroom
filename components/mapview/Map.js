import React, {useState, Component} from 'react';
import MapView, {PROVIDER_GOOGLE, Marker} from 'react-native-maps';
import { Image, View, StyleSheet } from 'react-native';
import icon from './icons/toilet.png'
class Map extends Component {

    state = {
        // Madison Wisconsin starting region
        region: {
            latitude: 43.07470053372165,
            longitude: -89.38453472496299,
            latitudeDelta: 0.0922,
            longitudeDelta: 0.0421,
        },
        markers: [
            {
                latlng: {
                    latitude: 43.0708751485606,
                    longitude:  -89.3986430824941
                },
                title: "Nicholas Recreation Center",
                description: "Pretty nice bathrooms"
            },
            {
                latlng: {
                    latitude: 37.78825,
                    longitude: -122.4324,
                    latitudeDelta: 0.0922,
                    longitudeDelta: 0.0421
                },
                title: "test",
                description: "a test marker"
            }
        ]
    }
    
    render () {
        return (
            <MapView
            style = {styles.container}
            region= {this.state.region}
            provider = {PROVIDER_GOOGLE}
            showsUserLocation = {true}
            >
                {/* Show all Markers */}
                {this.state.markers.map((markers, index) => {
                        return <Marker
                        key = {index}
                        coordinate = {markers.latlng}
                        title = {markers.title}
                        description = {markers.description}
                        style = {styles.marker}
                        // image = {{uri: Image.resolveAssetSource(icon).uri}}
                        />
                })}
            </MapView>
        );
    }
  };


const styles = StyleSheet.create({
    container: {
      ...StyleSheet.absoluteFillObject,
      flex: 1
    },
    marker: {
        height:50,
        width: 20
    }
  });

export default Map;