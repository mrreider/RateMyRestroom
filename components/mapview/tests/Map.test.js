import React, {Component} from 'react';
import Map from '../Map'
import Adapter from 'enzyme-adapter-react-16'
import {shallow, configure} from 'enzyme'
import toJson from 'enzyme-to-json'

// This configures jest to be able to work with enzyme
configure({adapter: new Adapter()})
// STATE TEST

/**
 * Creates an instance of map, runs addMarker,
 * then checks that the length of the markers list in state
 * has been incremented by one
 */
it('Should change the state of markers, incremented by one', () => {
    const c = shallow(<Map/>)
    const marker =  {
        latlng: {
            latitude: 43.0708751485606,
            longitude:  -89.3986430824941
        },
        title: "Nicholas Recreation Center",
        description: "Pretty nice bathrooms"
    }
    c.instance().addMarker(marker.latlng, marker.title, marker.description)
    expect(c.state('markers').length).toBe(3)
})

/**
 * Runs the onRegionChange function, which changes state
 * to be equal to its given parameter. Checks if
 * state is properly changed
 */
it('Should change the region in state', () => {
    const c = shallow(<Map/>)
    const newRegion = {
        latitude: 0,
        longitude: 0,
        latitudeDelta: 0,
        longitudeDelta: 0
    }
    c.instance().onRegionChange(newRegion)
    expect(c.state('region')).toEqual({
        latitude: 0,
        longitude: 0,
        latitudeDelta: 0,
        longitudeDelta: 0
    })
})

/**
 * Snapshot testing of map.
 * Tests from previous snapshots
 */
it ('Should match the previous snapshot', () => {
    const c = shallow(<Map/>)
    expect(toJson(c)).toMatchSnapshot()
})