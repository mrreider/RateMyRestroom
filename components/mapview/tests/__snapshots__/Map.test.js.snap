// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the previous snapshot 1`] = `
<MapView
  provider="google"
  region={
    Object {
      "latitude": 43.07470053372165,
      "latitudeDelta": 0.0922,
      "longitude": -89.38453472496299,
      "longitudeDelta": 0.0421,
    }
  }
  showsUserLocation={true}
  style={
    Object {
      "bottom": 0,
      "flex": 1,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <MapMarker
    coordinate={
      Object {
        "latitude": 43.0708751485606,
        "longitude": -89.3986430824941,
      }
    }
    description="Pretty nice bathrooms"
    image={1}
    key="0"
    stopPropagation={false}
    style={
      Object {
        "height": 50,
        "width": 20,
      }
    }
    title="Nicholas Recreation Center"
  />
  <MapMarker
    coordinate={
      Object {
        "latitude": 37.78825,
        "latitudeDelta": 0.0922,
        "longitude": -122.4324,
        "longitudeDelta": 0.0421,
      }
    }
    description="a test marker"
    image={1}
    key="1"
    stopPropagation={false}
    style={
      Object {
        "height": 50,
        "width": 20,
      }
    }
    title="test"
  />
</MapView>
`;
